### PR TITLE
isdefined check in locals()

### PR DESCRIPTION
**Pull request by pfitzseb**
~~Not sure how to best test this though.~~ Now with tests.

### CODE REVIEW

**Comment by timholy** [comment]
Presumably this fixed an error, do you have a reproducer?

**Comment by pfitzseb** [comment]
Yeah. Added a test...

File: test/interpret.jl [hunk 1]
```diff
@@ -640,3 +640,12 @@ end
     D = Diagonal([1.0, 2.0])
     @test @interpret(f(D)) === f(D)
 end
+
+@testset "constructor locals" begin
+    struct A
+        a::Int
+    end
+
+    frame = JuliaInterpreter.enter_call(A, 3)
```
Comment (timholy) [inline test/interpret.jl:649]:
```suggestion
    struct A396
        a::Int
    end

    frame = JuliaInterpreter.enter_call(A396, 3)
```

**Comment by timholy** [comment]
Looks like you've got a Julia 1.0-specific bug, otherwise this is good to go. (If you edit it note I added a commit that fixed a different problem.)

**Pull request merged**